fw/http_parser.c

```diff
@@ -6719,6 +6769,38 @@ __h2_req_parse_x_forwarded_for(TfwHttpMsg *hm, unsigned char *data, size_t len,
 }
 STACK_FRAME_NON_STANDARD(__h2_req_parse_x_forwarded_for);
 
+
+static int
+__h2_req_parse_x_tempesta_cache(TfwHttpMsg *hm, unsigned char *data, size_t len,
+			  bool fin)
+{
+	int r = CSTR_NEQ;
+	__FSM_DECLARE_VARS(hm);
+
+	__FSM_START(parser->_i_st);
+
+	__FSM_STATE(Req_I_X_Tempesta_Cache) {
+		/*
+		 * Tempesta custom header for PURGE:
+		 *	X-Tempesta-Cache = "X-Tempesta-Cache" ":" method
+		 *	method = "get"
+		 */
+		H2_TRY_STR("get", Req_I_X_Tempesta_Cache, I_Tempesta_Cache_get);
+		__FSM_EXIT(CSTR_NEQ);
+	}
+
+	__FSM_STATE(I_Tempesta_Cache_get) {
+		if (IS_CRLF(c)) {
+			// do some processing of PURGE X-Tempesta-Cache request
+			hm->cache_ctl.flags |= TFW_HTTP_CC_CACHE_PURGE;
+			return __data_off(p);
+		}
+		__FSM_EXIT(CSTR_NEQ);
```

Comment:
I can't help but notice that the code now has a mix of two ways/styles to exit. One is using `__FSM_EXIT()` macro, and the the other is a straight `return` statement. For instance, `__FSM_EXIT(CSTR_NEQ)` and `return CSTR_NEQ`. The two styles can even be seen mixed together in the same functions.

It's not this new code's issue, it's been this way for a while.